lib/tennis_game.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# TennisGame: used to manage the scoring in a tennis game
+class TennisGame
+  TENNIS_POINTS = %w(Love Fifteen Thirty Fourty).freeze
+  attr_accessor :points
+  def initialize
+    # create score array with scores for both teams
+    @points = [0, 0]
+  end
+
+  # converts the score into a the tennis scores
+  def score
+    if deuce?
+      'Deuce'
+    # if both teams have at least 3 points but are not equal
+    elsif advantage?
+      "Advantage Team #{which_team_advantage}"
+    elsif win?
+      "Team #{which_team_win} Win!"
+    else
+      # just show converted scores
+      "#{TENNIS_POINTS[@points[0]]} : #{TENNIS_POINTS[@points[1]]}"
+    end
```

Comment:
This is a good way we can work out the score. You can also write it like this: 

```
def score
   return 'Deuce' if deuce?
   return "Team #{which_team_win} Win!" if win?
   "#{TENNIS_POINTS[@points[0]]} : #{TENNIS_POINTS[@points[1]]}"
end
```
It means you can condense the logic into 3 lines. The caveat is that Advantage is moved into `TENNIS_POINTS`